lib/client.js

```diff
@@ -1331,6 +1331,65 @@ MatrixClient.prototype.invite = function(roomId, userId, callback) {
 };
 
 /**
+ * Invite a user to a room based on their email address.
+ * @param {string} roomId The room to invite the user to.
+ * @param {string} email The email address to invite.
+ * @param {string} displayName The display name to use. If omitted, everything
+ * before the '@' in the email address is used.
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.inviteByEmail = function(roomId, email, displayName, callback) {
+    if (utils.isFunction(displayName)) {
+        callback = displayName; displayName = undefined;
+    }
+    if (!displayName) {
+        displayName = email.split("@")[0];
+    }
+    return this.inviteByThreePid(
+        roomId, "email", email, displayName, callback
+    );
+};
+
+/**
+ * Invite a user to a room based on a third-party identifier.
+ * @param {string} roomId The room to invite the user to.
+ * @param {string} medium The medium to invite the user e.g. "email".
+ * @param {string} address The address for the specified medium.
+ * @param {string} displayName The human-readable name to show in the room for
+ * this invite. MUST NOT be the address.
+ * @param {string} idServer Optional. The identity server URL to use for inviting.
```

Comment:
Well the C/S API exposes which IS to use by necessity since it needs to know what server to go & hit, just as we need to know what IS to use when we hit the IS directly. The SDK offer the ability to change the IS used on a request-by-request basis anywhere else so I don't think this really needs to be any different.